compiler/src/iree/compiler/Codegen/Common/EncodingUtils.cpp

```diff
@@ -136,6 +138,43 @@ MaterializeEncodingTypeConverter::getEncodingInfo(RankedTensorType type) const {
   return layoutAttr.getEncodingInfo(type);
 }
 
+FailureOr<SmallVector<OpFoldResult>>
+MaterializeEncodingTypeConverter::getInnerTileSizesOfr(
+    OpBuilder &rewriter, Location loc, RankedTensorType tensorType,
+    const IREE::Codegen::MaterializeEncodingInfo &materializeEncodingInfo)
+    const {
+  ArrayRef<int64_t> staticTileSizes = materializeEncodingInfo.innerTileSizes;
+  if (llvm::all_of(staticTileSizes,
+                   [](int64_t i) { return !ShapedType::isDynamic(i); })) {
```

Comment:
We can avoid the lambda if we use `ShapedType::isDynamicShape(staticTileSizes)`. It also looks cleaner to me.

I understand that it is refactoring the code. I'd appreciate it if you update the old code during the refactoring. We can either fix it in my PR (https://github.com/iree-org/iree/pull/20731) or in your PR.

(My PR exclude this case to avoid conflict. I can make the change after you land this. Either works for me. I'll let you make the call.)